### PR TITLE
Move the example code from the README to an examples/ directory.

### PR DESCRIPTION
**Pull request by morrowc**
Also correct a few mistakes in the example... which now runs even.

### CODE REVIEW

File: rpsl-parser/examples/main.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		fmt.Printf("the initial character read(%v) is not a letter, file unparsable.", r)
+		return
+	}
+
+	// Start to Parse() the reader contents, report results up the channel (rc).
+	rc := make(chan *rpsl.Record)
+	go rpsl.Parse(rdr, rc)
+
+	for rec := range rc {
+		fmt.Printf("Record type(%v):\n", rec.Type)
```
Comment (aashaikh) [inline rpsl-parser/examples/main.go:49]: nit: could add a comment that this is the block where one would "process" the parsed records/fields

File: rpsl-parser/examples/main.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		fmt.Printf("Failed to open irrFile(%v): %v\n", irrFile, err)
+		return
+	}
+
+	// Make a new rpsl.Reader{}.
+	rdr := rpsl.NewReader(fd)
+
+	// Read the first character from the file.
+	// Use this character to validate that the file MIGHT be correct.
```
Comment (aashaikh) [inline rpsl-parser/examples/main.go:26]: is this check necessary, if you are anyway checking that the first char is a letter below?

Comment (morrowc) [reply rpsl-parser/examples/main.go:26]: This gets the character to read... it effectively reads 1 char, then unreads (resets the pointer to the where the read char was). then passes the read char (r) to the 'isletter' function.

File: rpsl-parser/examples/main.go (hunk 1 above)
Comment (morrowc) [reply rpsl-parser/examples/main.go:49]: totes! :)

File: rpsl-parser/examples/main.go (hunk 2 above)
Comment (aashaikh) [reply rpsl-parser/examples/main.go:26]: Ok, i guess you could avoid if you had Peek() method ...  the wrapper that defines ReadRune() behavior as Read() is a little confusing (to me).

**Review by aashaikh — APPROVED** [review]
LGTM.  May want to think a bit more about the rpsl reader API in a future revision.

File: rpsl-parser/examples/main.go (hunk 2 above)
Comment (morrowc) [reply rpsl-parser/examples/main.go:26]: There is a peek(), but it's not exported.
huh, I could just export it :) I'll think on that on the flight home.

Comment (morrowc) [reply rpsl-parser/examples/main.go:26]: I'll kick out an exported Peek() in the next bit.

**Pull request merged**